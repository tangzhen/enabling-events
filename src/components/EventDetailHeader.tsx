import { Typography } from "@material-ui/core";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import IconButton from "@material-ui/core/IconButton";
import Paper from "@material-ui/core/Paper";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import FavoriteIcon from "@material-ui/icons/Favorite";
import ShareIcon from "@material-ui/icons/Share";
import React from "react";
import { EnablingEvent } from "../components/HomeEventItem";

interface EventDetailHeaderProps {
  event: EnablingEvent;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    container: {
      marginTop: theme.spacing(4),
      marginBottom: theme.spacing(4),
      display: "flex",
      flexDirection: "column",
    },
    bg: {
      height: "400px",
      backgroundSize: "cover",
      backgroundRepeat: "no-repeat",
      backgroundPosition: "center",
    },
    organization: {
      display: "flex",
      alignItems: "center",
      flexWrap: "wrap",
    },
    title: {
      width: "100%",
    },
    followButton: {
      marginLeft: theme.spacing(3),
    },
    description: {
      padding: theme.spacing(4),
      display: "flex",
      flexDirection: "column",
      justifyContent: "space-between",
      backgroundColor: theme.palette.grey[50],
    },
    footer: {
      padding: theme.spacing(1),
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
    },
  })
);

export default function EventDetailHeader({ event }: EventDetailHeaderProps) {
  const classes = useStyles();

  return (
    <>
      <Paper className={classes.container}>
        <div>
          <Grid container>
            <Grid item xs={8}>
              <img src={event.bg} alt={event.title} className={classes.bg} />
            </Grid>
            <Grid item xs={4} className={classes.description}>
              <Typography>{event.date}</Typography>
              <div className={classes.organization}>
                <Typography variant="h6" className={classes.title}>
                  {event.title}
                </Typography>
                <Typography>by {event.org}</Typography>
                <Button
                  variant="outlined"
                  color="primary"
                  className={classes.followButton}
                >
                  FOLLOW
                </Button>
              </div>
              <Typography>FREE</Typography>
            </Grid>
          </Grid>
        </div>

        <div className={classes.footer}>
          <IconButton aria-label="add to favorites">
            <FavoriteIcon />
          </IconButton>
          <IconButton aria-label="share">
            <ShareIcon />
          </IconButton>
        </div>
      </Paper>
    </>
  );
}
