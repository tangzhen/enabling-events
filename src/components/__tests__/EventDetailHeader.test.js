import React from "react";
import { render } from "../../../test/test-utils";
import EventDetailHeader from "../EventDetailHeader";

describe("HomeEventCategory", () => {
  it("should render component correct", () => {
    const event = {
      id: 1,
      title: "Pancake With Sliced Strawberry",
      bg: "/files/1.jpg",
      brief:
        "A pancake (or hotcake, griddlecake, or flapjack) is a flat cake, often thin and round, prepared from a starch-based batter that may contain eggs, milk and butter.",
      date: "20190228",
      org: "Standard Chartered",
      location: "Entrance of National Gallery",
      description:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci sagittis eu volutpat odio facilisis mauris sit amet massa vitae tortor condimentum.",
    };
    const { container } = render(<EventDetailHeader event={event} />);
    expect(container).toMatchSnapshot();
  });
});
