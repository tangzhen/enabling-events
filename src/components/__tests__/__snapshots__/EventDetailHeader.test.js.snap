// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeEventCategory should render component correct 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <img
            alt="Pancake With Sliced Strawberry"
            class="makeStyles-bg-2"
            src="/files/1.jpg"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-description-6 MuiGrid-item MuiGrid-grid-xs-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            28 Feb 2019
          </p>
          <div
            class="makeStyles-organization-3"
          >
            <h6
              class="MuiTypography-root makeStyles-title-4 MuiTypography-h6"
            >
              Pancake With Sliced Strawberry
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              by 
              Standard Chartered
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-followButton-5 MuiButton-outlinedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                FOLLOW
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            FREE
          </p>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-footer-7"
    >
      <button
        aria-label="add to favorites"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="share"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
